cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -2399,33 +2414,45 @@ String getCertificateThumbprint(Secret certSecret, CertAndKeySecretSource custom
             }
         }
 
-        ConfigMap getKafkaAncillaryCm()    {
-            ConfigMap loggingCm = null;
+        Future<ConfigMap> getKafkaAncillaryCm()    {
+            Future<ConfigMap> metricsCm = Future.succeededFuture(null);
```

Comment:
Same as above ... create the succeeded future only when it is really needed. This might be also on other places.